src/components/Form/Form.md

```diff
@@ -354,6 +353,167 @@ const maxAge = 65
+          )
+        },
+        validators: [validateIllegal],
+        validationSuccess: [analyticsCustomEvent],
+        labelCopy: 'Toggle dynamic fields',
+        options: [
+          { value: 'toggle-1', copy: 'Toggle Field 1' },
+          { value: 'toggle-2', copy: 'Toggle Field 2' },
+        ],
+      },
+      field1: {
+        component: (props, options) => {
+          return <TextInput
+            placeholder = "field 1 depends on choice 1--enter something for another dynamic field"
+            {...props}
+          />
+        },
+        validators: [(val) => val.length > 2 ? '2 chars max allowed' : ''],
+        optional: true,
+        labelCopy:
+          "optional with validation",
+        tid: 'field-1-text-input',
+      },
+      field2: {
+        component: (props, options) => {
+          return <TextInput
+            placeholder = "field 2 depends on choice 2"
+            {...props}
+          />
+        },
+        validators: [(val) => val.length > 2 ? '2 chars max allowed' : ''],
+        optional: true,
+        labelCopy:
+          "optional with validation",
+        tid: 'field-2-text-input',
+      },
+      field3: {
+        component: (props, options) => {
+          return <TextInput
+            placeholder = "field 3 depends on input in field 1"
+            {...props}
+          />
+        },
+        validators: [(val) => val.length > 2 ? '2 chars max allowed' : ''],
+        optional: true,
+        labelCopy:
+          "optional with validation",
+        tid: 'field-3-text-input',
+      },
+    },
+    onSubmit: async (formData) => {
+      if (count++ % 2 === 0) {
+        throw new Error("API Issue (Try again,, it alternates success & failure)")
+      } else {
+        alert(
+          'form submission successful with values:' +
+            JSON.stringify(formData)
+        )
+      }
+    },
+  }}
+>
+  {(api) => {
+    const {
+      field,
+      getFieldErrors,
+      getFieldValues,
+      getFormErrorMessage,
+      getFormIsValid,
+      getFormInteractedWith,
+    } = api
+    const values = getFieldValues()
+    const errors = getFieldErrors()
+    return (
+      <div>
+        <TitleLarge.Serif.Book500>
+          Dynamic Fields
+        </TitleLarge.Serif.Book500>
+
+        <Spacer.H16 />
+
+        {!!getFormInteractedWith() && (
+          <>
+            <InfoMessage.Alert.Success>
+              {"Form interacted with."}
+            </InfoMessage.Alert.Success>
+          </>
+        )}
+
+        {getFormErrorMessage() && (
+          <>
+            <InfoMessage.Alert.Error>
+              {getFormErrorMessage()}
+            </InfoMessage.Alert.Error>
+          </>
+        )}
+
+        {field('buttonGroup')}
+
+        <Spacer.H16 />
+
+        {values.buttonGroup == "toggle-1" && field('field1')}
+        {values.buttonGroup == "toggle-2" && field('field2')}
+
+        <Spacer.H16 />
+
+        { !errors.field1 && values.field1 && field('field3')}
```

Comment:
This is the guy that's interesting due to the contrived `2 chars max allowed` on `field1`...maybe an edge case though